WordPress/gutenberg
# Social Links block: Add color classes so icon colors correctly reflect changes in Global Styles

Fixes https://github.com/WordPress/gutenberg/issues/36462, https://github.com/WordPress/gutenberg/issues/35480, and partially fixes https://github.com/WordPress/gutenberg/issues/46547
This PR supersedes https://github.com/WordPress/gutenberg/pull/42284, but props to @BE-Webdesign for kicking this off. 

## What? & Why?
Before this PR, social icon colors were essentially hardcoded as hex values. This implementation is problematic if a color is set to a value in the color palette, and then you change the color palette value in Global Styles, or you select a different style variation. The icon colors did not update to the new values. This issue is best illustrated in the videos below.

## How?
This PR adds the CSS classes that correspond to the selected color values. Therefore, if an icon is set to the color `base`, whenever `base` changes in Global Styles or in theme.json, the icon will update accordingly. 

That said, the hardcoded hex values are important, and this PR does not remove them. The CSS classes just take priority. If you just used CSS classes, if a user switched to a theme that did not support `base`, the icons would not display correctly. In this situation, the hardcoded hex values kick in. 

## Testing Instructions

1. Use TT3 or a similar theme. (Themes with style variations make this easy to test)
2. Insert a new Social Icons block and add a few icons. 
3. Set the icon color and icon background color to colors in the theme palette.
4. In Global Styles either change the theme colors or choose a different style variation. 
5. Confirm that the icons colors have been updated to reflect the changes, both in the Editor and on the front end. 

## Screenshots or screencast

**Before:** 

https://github.com/WordPress/gutenberg/assets/4832319/c0ce7e51-8f73-40fe-a8d1-cc4d5cc06089



**After:**

https://github.com/WordPress/gutenberg/assets/4832319/cf261a9d-1ee2-410e-9082-e6083d01fe9f
